services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusServiceITest.kt

```diff
@@ -1745,49 +1749,190 @@ class HakemusServiceITest(
+                            .setOrdererForContractor(founder.id)
+                    expectedCableReportAlluRequest =
+                        expectedJohtoselvityshakemusDataAfterSend.toAlluCableReportData(
+                            hakemus.hankeTunnus)
+                    expectedExcavationAnnouncementAlluRequest =
+                        expectedKaivuilmoitusDataAfterSend.toAlluExcavationNotificationData(
+                            hakemus.hankeTunnus)
+                    cableReportAlluId = alluId
+                    excavationAnnouncenemtAlluId = alluId + 1
+                    every { alluClient.create(expectedCableReportAlluRequest) } returns
+                        cableReportAlluId
+                    every { alluClient.create(expectedExcavationAnnouncementAlluRequest) } returns
+                        excavationAnnouncenemtAlluId
+                    justRun { alluClient.addAttachment(cableReportAlluId, any()) }
+                    justRun { alluClient.addAttachment(excavationAnnouncenemtAlluId, any()) }
+                    justRun { alluClient.addAttachments(cableReportAlluId, any(), any()) }
+                    justRun {
+                        alluClient.addAttachments(excavationAnnouncenemtAlluId, any(), any())
+                    }
+                    every { alluClient.getApplicationInformation(cableReportAlluId) } returns
+                        AlluFactory.createAlluApplicationResponse(cableReportAlluId)
+                    every {
+                        alluClient.getApplicationInformation(excavationAnnouncenemtAlluId)
+                    } returns
+                        AlluFactory.createAlluApplicationResponse(
+                            excavationAnnouncenemtAlluId,
+                            applicationId = DEFAULT_EXCAVATION_ANNOUNCEMENT_IDENTIFIER)
+                }
 
-            assertThat(response).all {
-                prop(Hakemus::alluid).isEqualTo(alluId)
-                prop(Hakemus::applicationIdentifier)
-                    .isEqualTo(ApplicationFactory.DEFAULT_APPLICATION_IDENTIFIER)
-                prop(Hakemus::alluStatus).isEqualTo(ApplicationStatus.PENDING)
-                prop(Hakemus::applicationData).isEqualTo(expectedDataAfterSend)
-            }
-            assertThat(hakemusRepository.getReferenceById(hakemus.id)).all {
-                prop(HakemusEntity::alluid).isEqualTo(alluId)
-                prop(HakemusEntity::applicationIdentifier)
-                    .isEqualTo(ApplicationFactory.DEFAULT_APPLICATION_IDENTIFIER)
-                prop(HakemusEntity::alluStatus).isEqualTo(ApplicationStatus.PENDING)
-            }
-            verifySequence {
-                alluClient.create(expectedAlluRequest)
-                alluClient.addAttachment(alluId, any())
-                alluClient.addAttachments(alluId, any(), any())
-                alluClient.getApplicationInformation(alluId)
+                @AfterEach
+                fun checkMocks() {
```

Comment:
This could use a comment explaining why it's here.

Reply:
I added comments for this and setUp functions both.